### PR TITLE
CFINSPEC-78: Add `linux_audit_system` resource

### PR DESCRIPTION
**Pull request by ahasunos**
✅ Signed-off-by: Sonu Saha [sonu.saha@progress.com](mailto:sonu.saha@progress.com)
## Related Issue
**CFINSPEC-78: Add `linux_audit_system` resource.**

## Description
- Using the `linux_audit_system` resource
Given that the user has called the `linux_audit_system` resource 
and given that the system is linux-based
then the resource allows to test configuration of linux audit system with new matchers and property.
- Property: `rules`
- Matchers: `be_enabled` and `be_running`
- Syntax
  ```
  describe linux_audit_system do
      it { should be_enabled }
      it { should be_running }
      its("rules") { should include "-w /etc -p wa" }
  end
  ```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: lib/inspec/resources/linux_audit_system.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    # Resource appearance in test reports.
+    def to_s
+      "linux_audit_system"
+    end
+
+    # The be_enabled matcher checks if the auditing is enabled.
+    # The enabled flag 1 indicates that the auditing is enabled.
+    def enabled?
+      auditctl_cmd = inspec.command("#{auditctl_utility} -s | grep enabled")
```
Comment (ahasunos) [inline lib/inspec/resources/linux_audit_system.rb:39]: Please let me know if we would like to change the way we fetch the report status information for checking enabled and pid values.

File: docs-chef-io/content/inspec/resources/linux_audit_system.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+## Examples
+The following examples show how to use this Chef InSpec audit resource.
+
+### Ensure auditing is enabled on the system
+
+`be_enabled` checks if auditing is enabled on the system.
+
+```ruby
+    describe linux_audit_system do
+      it { should be_purple }
```
Comment (clintoncwolfe) [inline docs-chef-io/content/inspec/resources/linux_audit_system.md]: be_purple?

**Review by clintoncwolfe — APPROVED** [review]

File: docs-chef-io/content/inspec/resources/linux_audit_system.md (hunk 2 above)
Comment (ahasunos) [reply docs-chef-io/content/inspec/resources/linux_audit_system.md]: 🤦🏽‍♂️ I will minimise such hiccups in the code.

**Review by dkumaras — COMMENTED** [review]
Pls check and confirm

**Comment by ahasunos** [comment]
> Pls check and confirm

Looks good to me.

**Pull request merged**